### PR TITLE
[MH-115] sitewide styles framework

### PR DESCRIPTION
**Pull request by nmashton**
This sets up the general approach to sitewide styles that we can use in future styling tickets, and it adds in some basic styles that will be shared across pages.

## The general approach

We are using a Bootstrap theme to define styles. In practice, this means:

- The Bootstrap CSS is included by means of a Sass import within `myhpom/static/myhpom/myhpom.scss`.
- To the extent that Bootstrap styles can be changed by setting the values of variables, those are set within `myhpom/static/myhpom/bootstrap_theme`. This is the approach outlined [in the docs](https://getbootstrap.com/docs/4.0/getting-started/theming/#variable-defaults).
- If it's not possible to change the appearance of the elements in the desired way, sitewide overrides are defined in `myhpom/static/myhpom/bootstrap_overrides`. The `_theme` and `_overrides` are imported before and after the Bootstrap Sass, respectively.
- If particular components need custom styles that can't be set by means of Bootstrap variables (in this PR, this includes the appearance of the small-window collapsing header), those rules are defined in the components' stylesheets themselves.

## Typography

Shane has generously provided us files for the fonts used in the site, saving us from having to use Typekit and so on. These files are associated with `font-family` declarations in `myhpom/static/myhpom/_typography.scss`. Where the sizes involved are ambiguous, I tried to associate them with fine-grained size values in a sensible way, pegging the `normal` value to the font called "Regular", `bold` to "Bold", etc.

## The nav / menu

This is clearly the fussiest part of the whole setup, requiring a large number of overrides and tweaks. This is because the menu's specified style is not very Bootstrap-y—which is fine, but it takes work.

The main thing to bear in mind is that the orange-on-blue appearance of the activated mobile dropdown menu calls for whitespace finagling and related changes to the opening and closing CSS transitions. These are spelled out in the comments in the related Sass file.

### CODE REVIEW

File: myhpom/static/myhpom/include/_header.scss [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+.myhpom-header__logo {
+    max-width: 12rem;
+}
+
+
+/*
+    Mobile menu toggle button styles
+
+    NOTE: many of the changes and tweaks here relate to our
+    desire to get the button to fill the available vertical
```
Comment (sharrisoncaktus) [inline myhpom/static/myhpom/include/_header.scss:50]: line 50 – do you mean "available horizontal space"?

It looks great, btw.

File: myhpom/templates/myhpom/includes/header.html [hunk 2]
```diff
@@ -1,23 +1,39 @@
...
+                {% endif %}
+                <li class="nav-item"><a class="nav-link myhpom-header__nav-link" href="">How It Works</a></li>
+                <li class="nav-item"><a class="nav-link myhpom-header__nav-link" href="">FAQs</a></li>
             </ul>
-            <ul class="navbar-nav">
+            <hr class="myhpom-header__menu-small-divider d-block d-sm-block d-md-none d-lg-none d-xl-none" />
+            <ul class="navbar-nav myhpom-header__navbar-nav">
                 {% if user.is_authenticated %}
-                <li class="nav-item"><a class="nav-link" href="{% url "myhpom:logout" %}">LOG OUT</a></li>
+                <li class="nav-item"><a class="nav-link myhpom-header__nav-link" href="{% url "myhpom:logout" %}">LOG OUT</a></li>
```
Comment (sharrisoncaktus) [inline myhpom/templates/myhpom/includes/header.html]: line 29 – title case (Log Out) as with surrounding links?

**Review by sharrisoncaktus — COMMENTED** [review]
It all looks good and works well in the interface. One little nitpick about LOG OUT.

File: myhpom/static/myhpom/include/_header.scss (hunk 1 above)
Comment (nmashton) [reply myhpom/static/myhpom/include/_header.scss:50]: I really did mean vertical—if I didn't make the changes here, the button would not stretch all the way from top to bottom of the nav bar.

**Pull request merged**